feat(ui): add page zoom controls

Serge45 opened the pull request:
As requested in #1256, I added zoom in, zoom out and reset zoom items to the edit menu. Two convenient tool buttons for zoom in and zoom out were added to browser tab as well. 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added zoom controls to the View menu (Zoom In, Zoom Out, Actual Size) with keyboard shortcuts
  * Introduced a per-tab zoom widget (toolbar menu) showing the current zoom percentage and offering Zoom In/Out and Reset
  * Zoom levels now update in real time across tabs so displayed percentages stay in sync
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Serge45 committed:
feat(ui): add menu button for zoom actions

An event filter was also added to prevent pop-up menu closing while
zoom in/out actions triggered.

trollixx commented:
Thanks for looking into this. I think adding more buttons to the toolbar results in UI clutter. Nowadays major browsers moved to one menu button. I'd say Zeal should follow the trend. That would also allow us to address #1251.

Would you be willing to give menu button try?

Serge45 commented:
Thanks, this is a good idea, I will move those buttons to a menu button ASAP.

trollixx reviewed (changes requested):
Looks good! I've added a few minor requests. Also please squash and rebase your changes. Thanks!

Serge45 commented on src/libs/ui/browsertab.cpp:
```diff
@@ -129,6 +130,31 @@ BrowserTab::BrowserTab(QWidget *parent)
+        m_webControl->resetZoom();
+    });
+
+    browserActionsMenu->installEventFilter(this);
```
Yes, the menu works expectedly without event filter. But without this event filter, user has to repeat click on tool button -> click on zoom button to zoom in/out multiple times if he/she wants to continuous zooming. I understand this is very different from other actions, I noticed that modern browsers like Chrome and FireFox implemented zoom actions in this way, so I mimic such behavior by event filter. If there is no need for this behavior, I can simply remove it :).

trollixx replied:
Ah, I see. Browsers actually do a bit different thing with +/- buttons embedded inside a menu item. I haven't tried but [`QWidgetAction`](https://doc.qt.io/qt-5/qwidgetaction.html) should allow us to do the same thing.

Serge45 replied:
Cool, looks like `QWidgetAction` is more reasonable to achieve this. I will start to refactor the menu with it. Would you like to wait or I can make another PR for this refactoring?

trollixx replied:
No rush. You can push the updated version to this PR.

Serge45 commented on src/libs/ui/mainwindow.ui:
```diff
@@ -91,6 +91,9 @@
      <string>&amp;Edit</string>
     </property>
     <addaction name="actionFind"/>
+    <addaction name="actionZoomIn"/>
```
Should I create this zoom menu in main menu bar or just add a nested menu in `Edit` menu?

trollixx replied:
Let's not make any changes to the main menu. I'll replace it with the new menu after merging your PR. The final UX will be similar to Chromium.

Serge45 replied:
Ok, thanks!

Serge45 committed:
feat(ui): use QWidgetAction to implement zoom actions

A custom QStyleSheet must be applied to BrowserZoomWidget
since QMenu does not handle item highlighting of QWidget
in QWidgetAction.

trollixx reviewed (commented):
Nice! Thank you! I have some nitpicks and small things I'd like to try. Is it okay if push my changes directly to this PR? I can also leave a detailed review but that might take longer.

Serge45 commented:
Ok, sure. There are some ugly codes to make widget in `QWidgetAction` behave like widget for `QAction` in `QMenu`, but I don't find a better way to implement it :(.

trollixx committed:
feat(ui): integrate zoom controls into View menu

Address review feedback from #1266 and update for current codebase:

- Add View → Zoom submenu (Zoom In, Zoom Out, Actual Size) with shortcuts; remove redundant QShortcut blocks
- Refactor BrowserZoomWidget to use QToolButton/autoRaise; drop QPushButton flat styling and setMouseTracking cargo
- Move per-tab ⋮ button to right of title label
- Use QWidgetAction without an event filter; expose signals from BrowserZoomWidget instead of widget pointers
- Add WebControl::zoomLevelChanged signal so the level label updates live
- Fix layout jitter by giving the level label a minimum width
- Add SPDX/copyright headers and Zeal::WidgetUi namespace to new files

Fixes #1256.

trollixx commented:
It took me less than 6 years to land this! 🤣 Thank you for the contribution again!
